loader_wsock32_proxy/CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+# loader_wsock32_proxy
+
+add_library(loader_wsock32_proxy SHARED
+            "dllmain.cpp"
+            "loader.cpp"
+            "loader.h"
+            "pch.cpp"
+            "pch.h"
+            "wsock32.asm"
+            "wsock32.def"
+)
+
+target_link_libraries(loader_wsock32_proxy PRIVATE
+                      ${CMAKE_CURRENT_SOURCE_DIR}/include/MinHook.x64.lib
+                      mswsock.lib
+                      ws2_32.lib
+                      Shlwapi.lib
+                      imagehlp.lib
+                      dbghelp.lib
+                      kernel32.lib
+                      user32.lib
+                      gdi32.lib
+                      winspool.lib
+                      comdlg32.lib
+                      advapi32.lib
+                      shell32.lib
+                      ole32.lib
+                      oleaut32.lib
+                      uuid.lib
+                      odbc32.lib
+                      odbccp32.lib
+)
+
+target_include_directories(loader_wsock32_proxy PRIVATE
+                           include
+)
+
+target_precompile_headers(loader_wsock32_proxy PRIVATE pch.h)
+
+target_compile_definitions(loader_wsock32_proxy PRIVATE
+                           UNICODE
+                           _UNICODE
+)
+
+set_target_properties(loader_wsock32_proxy PROPERTIES
+                      RUNTIME_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/game/bin/x64_retail
```

Comment:
Is `game` a common artifact folder name? Haven't interacted too much with CMake, I just know from past experience that one usually calls it `build` or `target` but I could be mistaken here ^^

Basically just wanna use an output folder name that is a standard as possible so that someone with CMake experience finds the build artifacts in the folder they expected.

Reply:
Cmake projects usually use folders named `bin` or `build` for output. I chose `game` mainly because r5reloaded uses the same structure

Reply:
cmake already has a concept of a build dir, so we shouldn't be overriding that IMO

Reply:
When using Visual studio as the generator CMAKE_BINARY_DIR defaults to `out/build/CMAKE_BUILD_TYPE/`. As I said i chose to output into `game` so that one can copy their game over and launch from vs directly.

I did try setting CMAKE_BINARY_DIR to `${CMAKE_SOURCE_DIR}/game`, but that seemed to default to the visual studio path ??, unsure what was happening there, might have been a problem somewhere else

Reply:
> When using Visual studio as the generator CMAKE_BINARY_DIR defaults to out/build/CMAKE_BUILD_TYPE/. As I said i chose to output into game so that one can copy their game over and launch from vs directly.

You should be able to set `buildRoot` in `CMakeSettings.json` to control that without breaking CMake conventions and the ability to do out-of-tree builds.

Reply:
> You should be able to set `buildRoot` in `CMakeSettings.json` to control that without breaking CMake conventions and the ability to do out-of-tree builds.

Due to cmake giving developers the ability to build in way more environments i chose to hard code the build path and ignore `CMakeSettings.json`

Reply:
What?

Reply:
> What?

Only the Visual Studio generator respects `CMakeSettings.json`. So we need to hard code the build dir into `CMakeLists.txt` to have cmake behave the same even when not being ran from visual studio.

Reply:
You pass the cmake build dir when setting up cmake.

Reply:
I'm fine with keeping NS_BINARY_DIR (it's good to have for the reason you mentioned), but I'd like it to respect CMake conventions by default.

Reply:
@F1F7Y

Reply:
Unsure what you mean by "respect CMake conventions". The reason I made `NS_BINARY_DIR` and not just used `CMAKE_BINARY_DIR` is because that is also used for the cache. That means setting `CMAKE_BINARY_DIR` to `${CMAKE_SOURCE_DIR}/game` will have cmake generate its files inside the game directory cluttering up your titanfall install

Reply:
When you do the following, you should end up with the binaries somewhere in the `./build` folder.

```
mkdir build
cd build
cmake /path/to/NorthstarLauncher
ninja
```

With `NS_BINARY_DIR` set to `${CMAKE_SOURCE_DIR}/game` as the default, the binaries end up in `/path/to/NorthstarLauncher/game` rather than `./build/game` like they would with `${CMAKE_BINARY_DIR}/game`.